src/model/topology.hpp

```diff
@@ -102,6 +102,25 @@ class Topology : public Module
  private:
   std::vector<std::shared_ptr<Level>> levels_;
   std::map<std::string, std::shared_ptr<Network>> networks_;
+
+  // Maps to store the binding relationship between architectural tiling level
+  // to actual micro-architecture. The key here is a temporal/spatial tiling
+  // level id, and the value is the pointer to the actual micro-architecture
+  // (buffer/network) for the tiling level. We might deprecate the above two
+  // members in the future.
+
+  // Level map
+  std::map<unsigned, std::shared_ptr<Level>> level_map_;
+  // Network map. The pair of network connecting two storage levels should
+  // share the same spatial tiling id. Note that these read_fill network and
+  // drain_update_network can be the same network.
+  struct Connection
+  {
+    std::shared_ptr<Network>  read_fill_network;
+    std::shared_ptr<Network> drain_update_network;
+  };
```

Comment:
Interesting. It seems this Connection map corresponds to a tileable target. As of today, we don't have a distinction between design-time connections between levels and config-time creation of user-(i.e.,mapping-)specified LI channels. When we do, do you think these Connections and the connection_map_ will refer to design-time connections or config-time channels?

Reply:
Currently they are used as design-time connections, similar as buffers in the topology (just like we don't have config-time creation of user-specified buffers). Moving forward, I think there is also a need for connection abstraction for physical network, too.